CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.6)
+
+set(CMAKE_CXX_FLAGS -Wall -std=c++17 -Werrors -O2)
```

Comment:
Better to use 
```cmake
target_compile_options(...)
```
Check CMake documentation. `O2` is already included into standard CMake flags in release mode.